src/dtproject/types.py

```diff
@@ -157,6 +157,15 @@ class DevContainerConfiguration(dict):
     service: Optional[str] = None
     workspaceFolder: Optional[str] = None
 
+@dataclasses.dataclass
+class Hook():
+    command: str = ""
+    required: Optional[bool] = False
+
+@dataclasses.dataclass
+class LayerHooks(DataClassLayer):
+    version: int
```

Comment:
`version` is only needed for serialization/deserialization, once in memory there is no need to store the version of the file.

Reply:
and BTW, we do have to add a 'version' field but we will have to call it `format` as some layers have the key `version` already taken (e.g., `self`, `format`, and `template`).

Reply:
Should this be the `format` of `FormatLayer` or indipendent?

Reply:
Remove `version`, add `format` to the schema to be used when parsing.